warehouse/query-core/src/main/java/datawave/query/predicate/TLDEventDataFilter.java

```diff
@@ -491,31 +491,37 @@ private Range getWhitelistSeek(Key current, String fieldName, Key endKey, boolea
         // none of the fields in the whitelist come after the current field
         if (range == null) {
             // roll to the next uid
-            range = getRolloverRange(current, endKey, endKeyInclusive, sortedWhitelist);
+            range = getRolloverRange(current, endKey, endKeyInclusive);
             lastListSeekIndex = -1;
         }
         
         return range;
     }
     
-    private Range getRolloverRange(Key current, Key end, boolean endInclusive, List<String> sortedWhitelist) {
-        Range range;
+    /**
+     * Get a rollover range that skips to the next child uid
+     *
+     * @param current
+     *            the current key
+     * @param end
+     *            the end key
+     * @param endInclusive
+     *            is end key inclusive flag
+     * @return a rollover range, or an empty range if the rollover range would extend beyond the end key
+     */
+    private Range getRolloverRange(Key current, Key end, boolean endInclusive) {
```

Comment:
It looks like the consequence of removing sortedWhitelist is that getRolloverRange will start at the next UID's 1st field, not the 1st field that appears in the sortedWhitelist - is that ok?

Reply:
Yes. The seek issued by the old code would only seek to the next child -- it ignored the field portion of the range.